File: day20/classwork/codewars.py
#Beginner - Lost Without a Map
def maps(a):
    new=[]
    for num in a:
        new.append(num*2)
    return new

#Keep Hydrated!
def litres(time):
    return int(time*0.5)

#Convert number to reversed array of digits
def digitize(n):
    new_arr=[]
    n=str(n)
    for i in n:
        new_arr.append(int(i))
    new_arr.reverse()
    return new_arr

#Basic Mathematical Operations
def basic_op(operator, value1, value2):
    if operator=="+":
        return value1+value2
    elif operator=="-":
        return value1-value2
    elif operator=="*":
        return value1*value2
    else:
        return value1/value2
    
#Century From Year
def century(year):
    if year%100==0:
        return year//100
    else:
        return year//100+1
    
#Beginner Series #2 Clock
def past(h, m, s):
    total= h * 3600000 + m * 60000 + s * 1000
    return total

#Beginner Series #1 School Paperwork
def paperwork(n, m):
    if n<0 and m<0:
        return 0
    elif n<0 or m<0:
        return 0
    else:
        return m*n
    
#Opposites Attract
def lovefunc( flower1, flower2 ):
    if (flower1%2==0 and flower2%2!=0) or (flower1 %2!=0 and flower2 %2==0):
        return True
    else:
        return False
    
#Abbreviate a Two Word Name
def abbrev_name(name):
    abrev=""
    abrev+=name[0]
    abrev+="."
    for i in range(len(name)):
        if name[i]==" ":
            abrev+=name[i+1]
    return abrev.upper()
#or
def abbrev_name(name):
    words = name.split()
    first_letter = words[0][0]
    first_letter.upper()
    second_letter= words[1][0]
    second_letter.upper()
    return "{}.{}".format(first_letter, second_letter)

#Simple multiplication
def simple_multiplication(number) :
    if number%2==0:
        return number*8
    else:
        return number*9

File: Day 57/Homework/codewars.js
// Returning Strings
function greet(name){
    return "Hello, " + name + " how are you doing today?"
}

// Convert a Boolean to a String
function booleanToString(b){
    if (b==true){
      return "true"
    } else{
      return "false"
    }
}

// Keep Hydrated!
function litres(time) {
  return (Math.floor(time*0.5));
}

// Basic Mathematical Operations
function basicOp(operation, value1, value2){
    if (operation=="+"){
      return value1+value2
    } else if (operation=="-"){
      return value1-value2
    } else if (operation=="*"){
      return value1*value2
    } else if (operation=="/"){
      return value1/value2
    }
}

// Century From Year
function century(year) {
    year=Math.floor((year+99) / 100);
    return year;
}

// Convert number to reversed array of digits
function digitize(n) {
  newArr=String(n).split("")
  res=[]
  for (let i=0; i<newArr.length; i++){
    res.push(Number(newArr[i]))
  }
  return res.reverse()
}

// Beginner - Lost Without a Map
function maps(x){
  res=[]
  for (let i=0; i<x.length; i++){
    res.push(x[i]*2)
  }
  return res
}

// Beginner Series #2 Clock
function past(h, m, s){
  milisec=h*3600000 + m*60000 + s*1000
  return milisec
}

// Opposites Attract
function lovefunc(flower1, flower2){
  if (flower1%2==0 && flower2%2!=0){
    return true
  } else if (flower1%2!=0 && flower2%2==0){
    return true
  } else{
    return false
  }
}

// Beginner Series #1 School Paperwork
function paperwork(n, m) {
  if (n<0 || m<0){
    res=0
    return res
  } else{
    return n*m
  }
}

// Abbreviate a Two Word Name
function abbrevName(name){
  newArr=name.split(" ");
  personName=newArr[0][0].toUpperCase();
  personLastName=newArr[1][0].toUpperCase();
  return `${personName}.${personLastName}`
}